L1Trigger/TrackFindingTracklet/src/VMRouterCM.cc

```diff
@@ -139,49 +143,91 @@ void VMRouterCM::execute() {
 
       allStubCounter++;
 
+      for (auto& allstub : allstubs_) {
+        allstub->addStub(stub);
+        if (settings_.debugTracklet()) {
+          edm::LogVerbatim("Tracklet") << getName() << " adding stub to " << allstub->getName();
+        }
+      }
+
       FPGAWord iphi = stub->phicorr();
       unsigned int iphipos = iphi.bits(iphi.nbits() - (settings_.nbitsallstubs(layerdisk_) + 3), 3);
 
-      //Fill allstubs memories - in HLS this is the same write to multiple memories
-      for (auto& allstub : allstubs_) {
+      unsigned int phicutmax = 4;
+      unsigned int phicutmin = 4;
+
+      if (layerdisk_ != 0) {
+        phicutmax = 6;
+        phicutmin = 2;
+      }
+
+      //Fill inner allstubs memories - in HLS this is the same write to multiple memories
+      for (auto& allstub : allinnerstubs_) {
         char memtype = allstub.first;
-        if ((memtype == 'R' && iphipos < 5) || (memtype == 'L' && iphipos >= 3) || (memtype == 'A' && iphipos < 4) ||
-            (memtype == 'B' && iphipos >= 4) || (memtype == 'E' && iphipos >= 4) || (memtype == 'F' && iphipos < 4) ||
-            (memtype == 'C' && iphipos >= 4) || (memtype == 'D' && iphipos < 4))
+        if (memtype == 'R' && iphipos < phicutmax)
+          continue;
+        if (memtype == 'L' && iphipos >= phicutmin)
+          continue;
+        if (memtype == 'A' && iphipos < 4)
+          continue;
+        if (memtype == 'B' && iphipos >= 4)
+          continue;
+        if (memtype == 'E' && iphipos >= 4)
           continue;
+        if (memtype == 'F' && iphipos < 4)
+          continue;
+        if (memtype == 'C' && iphipos >= 4)
+          continue;
+        if (memtype == 'D' && iphipos < 4)
+          continue;
+
+        int absz = std::abs(stub->z().value());
+        if (layerdisk_ == 1 && absz < 50.0 / settings_.kz(layerdisk_))
```

Comment:
what are the 50.0, 95.0, 55.0, 70.0 ?

Reply:
Cut on z or r position - replaced with more meaningful constants.